### PR TITLE
Adding Azure wrapper and its test cases

### PR DESCRIPTION
**Pull request by Mohannadcse**

### CODE REVIEW

**Comment by Mohannadcse** [comment]
I noticed for some reason when the `.env` contains `OPENAI_API_BASE`, immediately the app will assume Azure is used not OpenAI.

**Comment by Mohannadcse** [comment]
> Please add instructions in the README.md and in the setup section of docs

sure, I was contemplating that, but decided at this moment to keep the instructions inside the description of the class `AzureConfig`

**Comment by pchalasani** [comment]
Also need to add the azure related keys to the pytest workflow file, and then I'll add those as secrets in the repo

**Comment by Mohannadcse** [comment]
I added the instructions

File: docs/quick-start/setup.md [hunk 1]
```diff
@@ -119,6 +119,23 @@ GOOGLE_API_KEY=your-key
 GOOGLE_CSE_ID=your-cse-id
 ```
 
+### Microsoft Azure OpenAI setup
+In the root of the repo, copy the `.azure_env_template` file to a new file `.azure_env`: 
+
+```bash
+cp .azure_env_template .azure_env
```
Comment (pchalasani) [inline docs/quick-start/setup.md:126]: what about the other vars

File: docs/quick-start/setup.md [hunk 2]
```diff
@@ -119,6 +119,23 @@ GOOGLE_API_KEY=your-key
...
+```
+
+The file `.azure_env` contains four environment variables that are required to use Azure OpenAI: `AZURE_API_KEY`, `OPENAI_API_BASE`, `OPENAI_API_VERSION`, and `OPENAI_DEPLOYMENT_NAME`.
+
+Follow these steps from [Microsoft Azure OpenAI](https://learn.microsoft.com/en-us/azure/ai-services/openai/chatgpt-quickstart?tabs=command-line&pivots=programming-language-python#retrieve-key-and-endpoint) to find these variables.
```
Comment (pchalasani) [inline docs/quick-start/setup.md:131]: this should link to the env vars section, it's not linking there

**Review by pchalasani — APPROVED** [review]

**Pull request merged**